### PR TITLE
Crafting changes

### PR DESCRIPTION
**Pull request by ericmeek**
Made changes requested on my previous branch (craft-changes). Will be getting rid of that one.

### CODE REVIEW

File: profiles/base-crafting.yaml [hunk 1]
```diff
@@ -1,116 +1,385 @@
...
+    catalyst: coal nugget
+    catalyst-room: 8785
+    catalyst_number: 2
+    catalyst_volume: 10
+
+recipe_parts:
+  small padding:
+    Crossing:
+      part-room: 16667
+      part-number: 11
```
Comment (rpherbig) [inline profiles/base-crafting.yaml]: Are part numbers always the same in different cities?

File: profiles/base-crafting.yaml [hunk 2]
```diff
@@ -1052,7 +1331,7 @@ crafting_recipes:
   container: mortar
   herb1: georin
   herb1_stock: 7
-
+  herb2: na
```
Comment (rpherbig) [inline profiles/base-crafting.yaml]: Are these 'herb2' additions a merge issue?

File: remedy.lic [hunk 3]
```diff
@@ -5,17 +5,15 @@
 # ;remedy remedies 2 "blister cream" "red flower" nemoih bar mortar cream
 # ;remedy remedies 2 "blister cream" nemoih "not used" bar mortar cream  - for when only herb is required
 
-custom_require.call(%w(common common-crafting common-items))
```
Comment (rpherbig) [inline remedy.lic]: Are the changes to this file merge issues?

File: shape.lic [hunk 4]
```diff
@@ -81,10 +98,6 @@ class Shape
       command = "rub my #{@noun} with my rasp"
     when 'Applying the final touches'
       finish
-    when 'while it is inside of something', 'You fumble around but'
```
Comment (rpherbig) [inline shape.lic]: Are these removals merge issues?

File: smith.lic [hunk 5]
```diff
@@ -28,14 +28,14 @@ class Smith
     args = parse_args(arg_definitions)
 
     data = get_data('crafting')
-
-    discipline = data['blacksmithing']
+    settings = get_settings
+    discipline = data['blacksmithing'][settings.hometown]
+    @hometown = settings.hometown
     @container = get_settings.crafting_container
+    @bank_info = get_data('town')[settings.hometown]['deposit']['id']
```
Comment (rpherbig) [inline smith.lic]: I don't think `@bank_info` is used anywhere. Can it be removed?

File: workorders.lic [hunk 6]
```diff
@@ -176,8 +178,10 @@ class WorkOrders
         end
 
         order_item(info['stock-room'], info['stock-number'])
-        stow_hands
+        #stow_hands
```
Comment (rpherbig) [inline workorders.lic]: Can this commented out line be removed?

File: workorders.lic [hunk 7]
```diff
@@ -28,7 +28,9 @@ class WorkOrders
     settings = get_settings
     crafting_data = get_data('crafting')
     @bag = settings.crafting_container('backpack')
-    info = crafting_data[discipline]
+    @bank_info = get_data('town')[settings.hometown]['deposit']['id']
```
Comment (rpherbig) [inline workorders.lic]: I don't think `@bank_info` is used anywhere. Can it be removed?

File: workorders.lic [hunk 8]
```diff
@@ -217,6 +221,7 @@ class WorkOrders
         end
         order_item(info['stock-room'], info['stock-number'])
         stow_hands
+        buy_parts(recipe['part'], info['part-room']) || []
```
Comment (rpherbig) [inline workorders.lic]: What does `|| []` do here?

File: workorders.lic [hunk 9]
```diff
@@ -251,7 +264,8 @@ class WorkOrders
     stock_needed = ((quantity * recipe['volume'] - existing) / 100.0).ceil
     order_yarn(info['stock-room'], stock_needed, info['stock-number'])
 
-    find_sewing_room
+    data = get_data('crafting')['tailoring'][@hometown]
```
Comment (rpherbig) [inline workorders.lic]: I don't think `data` is used anywhere. Can it be removed?

File: workorders.lic [hunk 10]
```diff
@@ -288,76 +304,50 @@ class WorkOrders
...
     when 'What were'
       stow_tool('water')
     end
     stow_hands
 
+    stow_tool('water')
     # order herbs
+
     order_rem(info['stock-room'], quantity, recipe['herb1_stock'])
-    if recipe['herb2'].nil?
```
Comment (rpherbig) [inline workorders.lic]: Are these changes merge issues?

File: profiles/base-crafting.yaml (hunk 1 above)
Comment (ericmeek) [inline profiles/base-crafting.yaml]: Yes, so far.

File: workorders.lic [hunk 11]
```diff
@@ -357,7 +378,7 @@ class WorkOrders
 
     remaining_volume = 0
 
-    ensure_copper_on_hand(5000, @hometown)
+    ensure_copper_on_hand(5000)
```
Comment (rpherbig) [inline workorders.lic]: Were some intended changes lost here?

Comment (ericmeek) [inline workorders.lic]: Yes. Issues corrected and pushed.

File: crossing-repair.lic [hunk 12]
```diff
@@ -12,7 +12,7 @@ class CrossingRepair
   def initialize
     setup
 
-    ensure_copper_on_hand(@repair_withdrawal_amount, @bank_info)
```
Comment (rpherbig) [inline crossing-repair.lic]: Why was this changed?

Comment (ericmeek) [inline crossing-repair.lic]: Because crossing repair is only going to happen in Crossing and ensure_copper_on_hand defaults to Crossing.

Comment (rpherbig) [inline crossing-repair.lic]: The rest of crossing-repair uses hometown, so I think this call should also. We are (or are almost?) at the point we could just rename this to repair.lic.

Comment (ericmeek) [inline crossing-repair.lic]: How do you feel about having the ensure_copper_on_hand call in the setup section for now, using the string settings.hometown? The reason I ask is because in setup hometown is data from the get_data call, unlike how I've been using hometown (as the string).

After some of these major changes are implemented I'd like to work on consistency among the scripts in their structure and lookup calls.

Comment (rpherbig) [inline crossing-repair.lic]: Sounds good to me.

:+1: for consistency in a future PR. Let me know how I can help.

Comment (ericmeek) [inline crossing-repair.lic]: Will do. Changes in.

File: common-crafting.lic [hunk 13]
```diff
@@ -12,33 +12,49 @@ module DRCC
     ['nothing in there'].include?(DRC.bput('look in cruc', 'nothing in there', 'you see'))
   end
 
-  def find_empty_crucible
+  def find_empty_crucible(hometown)
```
Comment (rcuhljr) [inline common-crafting.lic]: Not sure this is the place for it, but could we just have all crucible rooms ever listed, and any that are over a certain distance are ignored? It's weird that I can't use ;smith without changing my hometown.

**Comment by rcuhljr** [comment]
Great work on the whole, big undertaking.
